src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -154,6 +154,22 @@ impl ScalarFunc {
         }
         Ok(Some(pow))
     }
+
+    #[inline]
+    pub fn truncate_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let x = try_opt!(self.children[0].eval_int(ctx, row));
+        let d = try_opt!(self.children[1].eval_int(ctx, row));
+        Ok(Some((truncate(x as f64, d)) as i64))
+    }
+}
+
+fn truncate(f: f64, d: i64) -> f64 {
```

Comment:
I will fix u64 issue later. this `truncate` help method is used for `truncateInt` and `truncateReal`. I will update the `truncate_int` implemention to what you describe. thanks for your review

Reply:
Ah, another issue, current implemention using the power trick,  so  we need cast i64 to f64, which will loose precision, the go version of  TiDB has the same problem. say a i64 number `9223372036854775807`. on TiDB, `select truncate(9223372036854773705, -3)` will result to `9223372036854773760`, on MySQL, will result to `9223372036854773000`.

we can use the `round` method of   `coprocessor::codec::mysql::Decimal` ,  for `truncate_int`, `truncate_real`, `truncate_decimal`, maybe this will be slower for `truncate_int` and `truncate_real`, but the result is correct. 
 any suggestions？ @BusyJay

Reply:
Why cast i64 to f64? I think it's OK to just divide/multiply an i64 by an i64.

Reply:
got it . forgot to use divide/multiply replace the old implemention. thanks